contracts/DaiBackstopSyndicate.sol

```diff
@@ -107,69 +106,56 @@ contract DaiBackstopSyndicate is DaiBackstopSyndicateInterface, SimpleFlopper, E
 
     // Ensure that sufficient Dai liquidity is currently available to withdraw.
     require(
-      daiRedeemed <= daiBalance, "Insufficient Dai (in use in auctions)"
+      daiRedeemed <= daiBalance, "DaiBackstopSyndicate/defect: Insufficient Dai (in use in auctions)"
     );
 
     // Redeem the Dai and MKR.
     _DAI_JOIN.exit(msg.sender, daiRedeemed);
-    require(_MKR.transfer(msg.sender, mkrRedeemed), "MKR redemption failed.");
+    require(_MKR.transfer(msg.sender, mkrRedeemed), "DaiBackstopSyndicate/defect: MKR redemption failed.");
   }
 
   /// @notice Triggers syndicate participation in an auction, bidding 50k DAI for 500 MKR
   /// @param auctionId ID of the auction to participate in
   function enterAuction(uint256 auctionId) external {
     require(
       block.timestamp >= _AUCTION_START_TIME,
-      "Cannot enter an auction before they have started."
+      "DaiBackstopSyndicate/enterAuction: Cannot enter an auction before they have started."
     );
 
     // Ensure that the auction in question has not already been entered
     require(
-      !_activeAuctions.contains(auctionId), "Already participating in this auction"
+      _bidders[auctionId] == address(0x0), 
+      "DaiBackstopSyndicate/enterAuction: Already participating in this auction"
     );
 
     // Determine the Dai currently held by the contract.
+    // ??? Probably not necessary since this should throw in the _bid() call
     uint256 daiBalance = _DAI.balanceOf(address(this));
     require(
-      daiBalance >= 50000 * 1e18, "Insufficient Dai available for auction."
+      daiBalance >= 50000 * 1e18, 
+      "DaiBackstopSyndicate/enterAuction: Insufficient Dai available for auction."
     );
 
-    // Ensure that the current bid is not at a higher price than backstop.
-    (uint256 currentDaiBid, uint256 curentLotSize, , , ) = SimpleFlopper.getCurrentBid(auctionId);
+    // Create auction's Bidder contract and approve it for VAT 
+    Bidder bidder = new Bidder(SimpleFlopper.getFlopperAddress(), auctionId);
+    _bidders[auctionId] = address(bidder);
+    _VAT.hope(address(bidder));
```

Comment:
@samczsun what's the plain-english explanation of what this one does?

Reply:
> We create our Auction Vault and enable it to pull DAI from the syndicate in order to bid

Here's my second attempt :upside_down_face: